explainaboard/metrics/meta_evaluation.py

```diff
@@ -120,14 +123,16 @@ def _calc_metric_from_aggregate_single(self, single_stat: np.ndarray) -> float:
         config = narrow(CorrelationNLGConfig, self.config)
         corr_func = config.get_correlation_func(config.correlation_type)
         if config.group_by == "dataset":
-            val = corr_func(single_stat[:, 0], single_stat[0:, 1])[0]
+            res = corr_func(single_stat[:, 0], single_stat[0:, 1])[0]
+            val = 0 if math.isnan(res) else res
```

Comment:
The same logic is implemented multiple times. Introducing a function is better as follows:
```python
def replace_nan(value: float, default: float) -> float:
    """docstring"""
    return default if math.isnan(value) else value
```